packages/mg-imagescraper/lib/ImageScraper.js

```diff
@@ -57,6 +59,16 @@ class ImageScraper {
         try {
             // Timeout after 20 seconds
             let response = await this.fetchImage(src);
+
+            // Get the content type from response headers and convert to extension
+            let contentType = response.headers['content-type'];
+            let ext = mime.getExtension(contentType);
+
+            // Replace the extension with what we know to be true from the headers
+            imageOptions.filename = replaceExt(imageOptions.filename, '.' + ext);
+            imageOptions.storagePath = replaceExt(imageOptions.storagePath, '.' + ext);
+            imageOptions.outputPath = replaceExt(imageOptions.outputPath, '.' + ext);
+
```

Comment:
My suggestion here is to:
- not run this for every file, but only for the ones where we don't have an extension available
- maybe try to use as much native Node.js as possible. We try to keep external dependencies to a minimum. In this case I think you can use the [`path` API ](https://nodejs.org/api/path.html#path_path)in Node.js - `path.extname(path)` and `path.format(pathObject)` could probably do the trick for our use-case 🤔

Reply:
Thanks for the tips :) Updated to use no dependencies. Given we're only allowing a few specific file types, we can use the mime to infer the extension.

I have have used it wrong, but I couldn't see how `path.format(pathObject)` could change the extension. It would work if the `src` itself was used here, but as the `imageOptions` object gets a default extension applied earlier, I think this is the best (quick) option.

This could probably be added to the laundry list for a larger refactor.

Reply:
I didn't try it myself yet, but can see this example in the [Node.js docs](https://nodejs.org/docs/latest-v12.x/api/path.html#path_path_format_pathobject):

```
// `name` + `ext` will be used if `base` is not specified.
path.format({
  root: '/',
  name: 'file',
  ext: '.txt'
});
```

is this not working in our case?

Reply:
I did try, but the extension didn't change. Maybe I added the wrong parameters? I'll have another go.

Reply:
If it doesn't work like that, feel free to merge this PR yourself as is. Looking good to me 👍